src/protocols/secure_channel/SessionEstablishmentDelegate.h

```diff
@@ -49,7 +51,9 @@ class DLL_EXPORT SessionEstablishmentDelegate
      */
     virtual void OnSessionEstablished(const SessionHandle & session) {}
 
-    virtual ~SessionEstablishmentDelegate() {}
+    // Triggered when the PairingSession has done its work (either finished or encountered an error), such that the PairingSession
+    // object can be release, and so does other resources associated to it.
+    virtual void OnSessionEstablishmentDone(PairingSession * pairing) = 0;
```

Comment:
We need to document that this will always immediately follow OnSessionEstablished or OnSessionEstablishmentError.

That said, I am left wondering why we have three separate callbacks of which we always call two in a way that requires consumers to store state.  It seems like a single "SessionEstablishmentDone" callback that is passed the PairingSession and the error, if any.  The consumer can then get the SecureSession from the PairingSession as desired...  It seems like it would be a lot simpler and less error-prone.

Also, it looks like PASESession was not updated to call `OnSessionEstablishmentDone`, which is a weird behavior difference between PASESession and CASESession that we should not have... again, an argument for having a simpler callback setup where that can't happen.

Reply:
Great idea, will follow this idea to use a single callback `OnSessionEstablishmentDone`